devicetree: Add DT_INST_GPARENT

pdgendt opened the pull request:
Add the instance-based version of DT_GPARENT.

pdgendt committed:
devicetree: Add DT_INST_GPARENT

Add the instance-based version of DT_GPARENT.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

galak reviewed (changes requested):
Needs a test added

gmarull commented on include/zephyr/devicetree.h:
```diff
@@ -382,6 +382,13 @@
+ * @param inst instance number
+ * @return a node identifier for the instance's grandparent
+ */
+#define DT_INST_GPARENT(inst) DT_GPARENT(DT_DRV_INST(inst))
```
all instance based macros need to be put into `devicetree-inst` group.

pdgendt committed:
devicetree: Move DT_INST_PARENT to devicetree-inst group

The DT_INST_PARENT macro belongs into the devicetree-inst
documentation group.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
tests: devicetree: add tests for DT_GPARENT/DT_INST_GPARENT

Add some coverage for the DT_GPARENT/DT_INST_GPARENT macros.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt commented:
> Needs a test added

@galak a test has been added, can you revisit the review?
